### PR TITLE
Load TrackerData lazily

### PR DESCRIPTION
**Pull request by mallexxx**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:
Task/Issue URL: https://app.asana.com/0/414709148257752/1202407381883565/f
GH Issue URL: https://github.com/duckduckgo/iOS/issues/1200
iOS PR:  https://github.com/duckduckgo/iOS/pull/1201
macOS PR: https://github.com/more-duckduckgo-org/macos-browser/pull/796
What kind of version bump will this require?: Major

**Optional**:

CC: @bwaresiak  @miasma13 

**Description**:
TrackerData was loaded synchronously using JSON Decoding notoriously known for its slowness which in rare cases led to termination at startup because of too long app initialisation
TrackerData initialisation was moved to Content Blocking worker queue, this should fix the issue

**Steps to test this PR**:
1. Validate Content Blocking Rules work without issues and no data races may happen

**OS Testing**:

* [ ] iOS
* [ ] macOS

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by bwaresiak** [comment]
This looks good on a first glance, but I need a bit more time to wrap my mind around this and make sure there are no corner cases and that it solves the original problem in https://github.com/duckduckgo/iOS/issues/1200 - let's add it to a next week's release instead of rushing this Today.

**Comment by bwaresiak** [comment]
I'll come back to it once current BSK-related work is wrapped up.

File: Sources/BrowserServicesKit/PrivacyConfig/PrivacyConfigurationManager.swift [hunk 1]
```diff
@@ -27,9 +27,13 @@ public protocol EmbeddedDataProvider {
 }
 
 public protocol PrivacyConfigurationManaging {
-    
+    var embeddedConfigData: PrivacyConfigurationManager.ConfigurationData { get }
+    var fetchedConfigData: PrivacyConfigurationManager.ConfigurationData? { get }
```
Comment (bwaresiak) [inline Sources/BrowserServicesKit/PrivacyConfig/PrivacyConfigurationManager.swift:31]: I wonder if we can get rid of that? Based on the fact that reload(nil) should fetch embedded, and reload(etag, data) would use provided one, maybe we can simply tweak tests so these become private?

File: Sources/BrowserServicesKit/UserScript/UserContentController.swift [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    where Pub: Publisher, Content: UserContentControllerNewContent, Pub.Output == Content, Pub.Failure == Never {
+
+        self.privacyConfigurationManager = privacyConfigurationManager
+        super.init()
+
+        cancellable = assetsPublisher.receive(on: DispatchQueue.main)
+            .map(ContentBlockingAssets.init)
+            .assign(to: \.contentBlockingAssets, onWeaklyHeld: self)
+
+#if DEBUG
```
[the remark this answers is not in the document]
Comment (mallexxx) [reply Sources/BrowserServicesKit/UserScript/UserContentController.swift:87]: it is; just to make it more explicit I've added extra flag for Debug builds in Package.swift

**Review by bwaresiak — APPROVED** [review]
LGTM!

**Pull request merged**